nltk/tokenize/casual.py

```diff
@@ -156,7 +156,7 @@
 
     # Remaining word types:
     r"""
-    (?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
+    (?:\w[\w'\-_]+\w)              # Words with apostrophes or dashes.
```

Comment:
Note that `\w` [includes](https://docs.python.org/2/library/re.html#regular-expression-syntax) all alphanumeric characters in the locale, plus `[0-9_]`. So the underscore here, and many of the underscores elsewhere in the file, are redundant.

If the intention is to match alphabetic characters only, `[^\W\d_]` is one solution: http://stackoverflow.com/a/12349512